src/apps/yield-protocol/arbitrum/yield-protocol.balance-fetcher.ts

```diff
@@ -0,0 +1,210 @@
+
+@Register.BalanceFetcher(YIELD_PROTOCOL_DEFINITION.id, network)
+export class ArbitrumYieldProtocolBalanceFetcher implements BalanceFetcher {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(YieldProtocolContractFactory) private readonly yieldProtocolContractFactory: YieldProtocolContractFactory,
+  ) {}
+
+  private async getBorrowBalances(address: string) {
+    const data = await this.appToolkit.helpers.theGraphHelper.request<YieldVaultRes>({
+      endpoint: yieldV2ArbitrumSubgraph,
+      query: vaultsQuery,
+      variables: { address },
+    });
+    const vaultOwner = data.vaultOwner;
+
+    if (!vaultOwner) return [];
+
+    const positions = await Promise.all(
+      vaultOwner.vaults.map(async vault => {
+        const {
+          debtAmount,
+          collateralAmount,
+          series: {
+            baseAsset: { id: artAddress, assetId: baseId },
+            fyToken: { maturity },
+          },
+          collateral: {
+            asset: { id: ilkAddress, assetId: ilkId },
+          },
+        } = vault;
+
+        // get the corresponding art (debt) and ilk (collateral) of the vault
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+        const art = baseTokens.find(v => v.address === artAddress.toLowerCase());
+        const ilk = baseTokens.find(v => v.address === ilkAddress.toLowerCase());
+
+        if (!art || !ilk) return null;
+
+        // data props
+        const collateral = drillBalance(ilk, parseUnits(collateralAmount.toString(), ilk.decimals).toString());
+        const debt = drillBalance(art, parseUnits(debtAmount.toString(), art.decimals).toString(), { isDebt: true });
+        const tokens = [collateral, debt];
+        const balanceUSD = sumBy(tokens, v => v.balanceUSD);
+        const collateralizationRatio = `${(debt.balanceUSD === 0
+          ? 0
+          : (collateral.balanceUSD / Math.abs(debt.balanceUSD)) * 100
+        )
+          .toFixed(2)
+          .toString()}%`;
+
+        const multicall = this.appToolkit.getMulticall(network);
+        const cauldron = this.yieldProtocolContractFactory.cauldron({ address: CAULDRON, network });
+        const [{ dec: cauldronPairDecimals }, { ratio }] = await Promise.all([
+          multicall.wrap(cauldron).debt(baseId, ilkId),
+          multicall.wrap(cauldron).spotOracles(baseId, ilkId),
+        ]);
+        const minCollatRatio = ratio / 10 ** cauldronPairDecimals;
+        const minCollateralizationRatio = `${(minCollatRatio * 100).toString()}%`;
+        const liqPrice = (debtAmount * minCollatRatio) / collateralAmount;
+        const liquidationPrice = `1 ${ilk.symbol}: ${liqPrice.toFixed(2)} ${art.symbol}`;
+
+        // display props
+        const displayProps: DisplayProps = {
+          label: `Yield Vault`,
+          secondaryLabel: `${getLabelFromToken(art)} Debt and ${getLabelFromToken(ilk)} Collateral`,
+          tertiaryLabel: formatMaturity(maturity),
```

Comment:
I just remembered that for `ContractPosition` types, we currently only show the `label` and not the `secondaryLabel` or `tertiaryLabel` in the UI.

It might be crowded, but try to fit all the important information into the `label` for now, until we iterate to show the `secondaryLabel` and `tertiaryLabel` on the UI.

Reply:
updated